fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/ConcatCursor.java

```diff
@@ -218,6 +220,20 @@ private ConcatCursorContinuation(boolean secondCursor, @Nonnull RecordCursorResu
             }
         }
 
+        @Override
+        @Nonnull
+        public ByteString toByteString() {
+            if (isEnd()) {
+                return ByteString.EMPTY;
+            } else {
+                //form bytes exactly once
+                if (cachedByteString == null) {
+                    cachedByteString = continuationFunction.apply(nextResult.getContinuation().toBytes()).toByteString();
```

Comment:
I think `continuationFunction` can now be redefined to be `Function<ByteString, RecordCursorProto.ConcatContinuation>`, and then one can call `nextResult.getContinuation.toByteString()` instead of `.toBytes()`. I _think_ that can actually save on copying byte arrays, at least sometimes.

There are a few other places in the continuations that currently call `ByteString.copyFrom(someContinuation.toBytes())` that can also probably be cleaned up.

Reply:
Fixed the ```continuationFunction``` in this way for both ```ConcatCursorContinuation``` and ```SizeStatisticsCollectorCursorContinuation```. @alecgrieser